### DISCUSSION TITLE
Handling data re-fetching in Nuxt 3 pages

### DISCUSSION THREAD

**Discussion by Gonzalo-Bruna** [nuxt/nuxt#29597] [Questions]
### Context
I'm currently working on a Nuxt 3 project where I have a component that fetches upcoming and past sessions for a user. The data is retrieved using the `useFetch` hook. However, I've encountered an issue where the data does not re-fetch when navigating back to the route after visiting another route. 

### Current Implementation
Here's a simplified version of my component's data fetching logic:

```ts
const { data: futureSessions, status: futureSessionsStatus } = useFetch<SessionsResponse>(
    `${runtimeConfig.public.apiBase}/professional/future-sessions/${userStore.user?.user_id}`,
    {
        method: "GET",
        credentials: "include",
        lazy: true,
    },
);

const { data: pastSessions, status: pastSessionsStatus } = useFetch<SessionsResponse>(
    `${runtimeConfig.public.apiBase}/professional/past-sessions/${userStore.user?.user_id}`,
    {
        method: "GET",
        credentials: "include",
        lazy: true,
    },
);
```

What options do I have to ensure the data is refreshed upon navigating back to this route? In the past, I've used a pattern like this:

```ts
watchEffect(() => {
    route.path; // Watching the route for changes
    refreshFutureSessions(); // Function to re-fetch data
});
```

However, I’m questioning if this is the best approach. Is there a more optimal way to handle data re-fetching in this scenario?

**Comment by AloisSeckar** [comment]
1. Along with the `data` and `status`, you can also get built-in `refresh` function from `useFetch` - which is designed to trigger data fetching again on request
2. First thing comming on my mind - I would try to place the invocation of `refresh` function into `onMounted` hook of the page component associated with the route
